File: src/main/java/it/contrader/dto/FindMistakeDTO.java
//Created By @Alessandro Alfieri
package it.contrader.dto;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIdentityInfo(generator=ObjectIdGenerators.PropertyGenerator.class, property="id")
public class FindMistakeDTO {
	
	private static final String typeGame = "FindMistake";
		
	private Long id;
	
	private String solution;
	
	private String definition;
	
	private String sentence;
	
	private String optionA;
	
	private String optionB;
	
	private String optionC;
	
	private CategoryDTO category;
	
	private LevelDTO level;
	
	public static String getTypeGame() {
		return typeGame;
	}
}

File: src/dto/findmistakedto.ts
//Created By @Alessandro Alfieri
import {LevelDTO} from './leveldto'
import {CategoryDTO} from './categorydto'

export class FindMistakeDTO{

    typeGame: "FindMistake";

    id: number;

    solution: string;
    
    definition: string;

    optionA: string;
    
    optionB: string;
    
    optionC: string;

    sentence: string;
    
    category: CategoryDTO;

    level: LevelDTO;

    categoryTitle: string;
    levelName: string;
}